### DISCUSSION TITLE
Autocomplete SelectedItem template

### DISCUSSION THREAD

**Discussion by martintamb12** [primefaces/community#828] [PrimeNG] [answered]
Hi, 

It makes no sense for me to not have a selectedItem template for the Autocomplete in single mode : https://www.primefaces.org/primeng-v14-lts/autocomplete. When I choose it, I have a pretty print of my object, but then when it's chosen, I have a very ugly UUID. I don't see how I can bypass this! 

Could you enable the template also for single mode?

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

We've investigated the issue and it seems that selectedItem template is only available in multiple mode because of the component design. In multiple mode, input box turns into an `<ul></ul>` instead of `<input/>` in single mode. List elements made templating possible in autocomplete, but we cannot provide a templating option for input value because component uses `<input/>` element. By selecting an option in single mode you'll see an input and the selectedItem display is actually an input value not an HTML element, in multiple mode you're receiving list items which is possible to implement templating. 

To make it possible, we need to redesign the component in single, replace the input by span element which will disable inputting a value to the input element. Right now we're implementing passthrough and we're postponing this request and add it into our internal backlog.

Thanks for reporting the issue!
